### PR TITLE
Improvements from upstream, bump plugwise to v0.31.8

### PR DESCRIPTION
**Pull request by CoMPaTech**
Following https://github.com/plugwise/python-plugwise/pull/359

### CODE REVIEW

**Comment by CoMPaTech** [comment]
Took the approach across binary_sensors, sensors and switch - seems to work, obviously annoying test-pypi validation but also two new typings in number and select.

**Comment by CoMPaTech** [comment]
Not done, but maybe we should first validate approach before spending too much time

File: custom_components/plugwise/models.py [hunk 1]
```diff
@@ -23,49 +22,30 @@
     UnitOfVolume,
     UnitOfVolumeFlowRate,
 )
-from plugwise import SmileBinarySensors, SmileSensors, SmileSwitches
+from plugwise.constants import BinarySensorType, SensorType, SwitchType
```
Comment (bouwew) [inline custom_components/plugwise/models.py:25]: These constants should be in the pw-beta `constants.py` file for now.
We can move them later to the backend if needed.

File: custom_components/plugwise/binary_sensor.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ async def async_setup_entry(
         config_entry.entry_id
     ][COORDINATOR]
 
-    entities: list[PlugwiseBinarySensorEntity] = []
+    entities: list[BinarySensorEntity] = []
```
Comment (bouwew) [inline custom_components/plugwise/binary_sensor.py:35]: This change is wrong, the previous type `PlugwiseBinarySensorEntity` is correct.

File: custom_components/plugwise/models.py (hunk 1 above)
Comment (bouwew) [reply custom_components/plugwise/models.py:25]: Never mind, hadn't seen https://github.com/plugwise/python-plugwise/pull/359

File: custom_components/plugwise/binary_sensor.py (hunk 2 above)
Comment (bouwew) [inline custom_components/plugwise/binary_sensor.py:35]: Please revert this change, what was there is correct.

Comment (CoMPaTech) [reply custom_components/plugwise/binary_sensor.py]: Search-replace failure :(

**Review by bouwew — APPROVED** [review]
The updates as suggested by Core dev's look good!

**Review by CoMPaTech — COMMENTED** [review]
👍 LGTM
